File: 2310-minimum-operations-to-halve-array-sum/2310-minimum-operations-to-halve-array-sum.cpp
class Solution {
public:
    int halveArray(vector<int>& nums) {
         priority_queue<long double> pq;
        long double sum = 0;
        
        for (const auto& it : nums) {
            sum += it;
            pq.push(it);
        }

        long double target = sum / 2.0;
        long double newSum = sum;
        int operations = 0;

        while (newSum > target) {
            long double element = pq.top();
            pq.pop();
            long double half = element / 2.0;
            newSum -= half;
            pq.push(half);
            operations++;
        }

        return operations;
    
    }
};

File: 2310-minimum-operations-to-halve-array-sum/2310-minimum-operations-to-halve-array-sum.java
class Solution {
    public int halveArray(int[] nums) {
             PriorityQueue<Double> pq = new PriorityQueue<>(Collections.reverseOrder());
        for(int num : nums){
            pq.add((double)num);
        }
        long sum = 0;
        for(int num : nums){
            sum += num;
        }
        int operation = 0;
        double ans = 0;
       while(ans <= (double) sum/2.0){
        operation++;
        double maxheap =pq.poll();
        double div = maxheap/2.0;
        ans += div;
         pq.add(div);
         if(ans==(double)sum/2.0) break;


       }
       return operation;
    }
}